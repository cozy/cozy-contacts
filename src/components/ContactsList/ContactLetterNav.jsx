import React from 'react'
import PropTypes from 'prop-types'

const ContactLetterNav = ({ letters, executeScroll }) => {
  letters.splice(letters.findIndex(letter => letter === 'EMPTY'), 1)
  return (
    <ul className="ContactLetterNav">
      {letters.map((letter, i) => {
        return (
          <li
            key={i}
            onClick={() => executeScroll(letter)}
            className="ContactLetterNav-letter"
          >
            {letter}
          </li>
        )
      })}
    </ul>
  )
}

ContactLetterNav.prototype = {
  letters: PropTypes.array.isRequired,
  executeScroll: PropTypes.func
}

export default ContactLetterNav
