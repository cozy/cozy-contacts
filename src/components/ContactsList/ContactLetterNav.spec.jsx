import React from 'react'

import AppLike from '../../tests/Applike'

import ContactLetterNav from './ContactLetterNav'
import renderer from 'react-test-renderer'
describe('ContactLetterNav', () => {
  test('Should match the Letter nav snapshot', () => {
    const letters = [
      'EMPTY',
      '(',
      '#',
      'A',
      'B',
      'C',
      'D',
      'E',
      'F',
      'G',
      'H',
      'J',
      'K',
      'L',
      'M',
      'N',
      'O',
      'P',
      'R',
      'S',
      'T',
      'V',
      'W',
      'Y'
    ]
    const letterNavInstance = (
      <AppLike>
        <ContactLetterNav letters={letters} executeScroll={() => {}} />
      </AppLike>
    )
    const tree = renderer.create(letterNavInstance).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
