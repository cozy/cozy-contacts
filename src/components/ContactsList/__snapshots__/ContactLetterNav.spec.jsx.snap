// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactLetterNav Should match the Letter nav snapshot 1`] = `
<ul
  className="ContactLetterNav"
>
  <li
    className="ContactLetterNav-letter"
    onClick={[Function]}
  >
    (
  </li>
  <li
    className="ContactLetterNav-letter"
    onClick={[Function]}
  >
    #
  </li>
  <li
    className="ContactLetterNav-letter"
    onClick={[Function]}
  >
    A
  </li>
  <li
    className="ContactLetterNav-letter"
    onClick={[Function]}
  >
    B
  </li>
  <li
    className="ContactLetterNav-letter"
    onClick={[Function]}
  >
    C
  </li>
  <li
    className="ContactLetterNav-letter"
    onClick={[Function]}
  >
    D
  </li>
  <li
    className="ContactLetterNav-letter"
    onClick={[Function]}
  >
    E
  </li>
  <li
    className="ContactLetterNav-letter"
    onClick={[Function]}
  >
    F
  </li>
  <li
    className="ContactLetterNav-letter"
    onClick={[Function]}
  >
    G
  </li>
  <li
    className="ContactLetterNav-letter"
    onClick={[Function]}
  >
    H
  </li>
  <li
    className="ContactLetterNav-letter"
    onClick={[Function]}
  >
    J
  </li>
  <li
    className="ContactLetterNav-letter"
    onClick={[Function]}
  >
    K
  </li>
  <li
    className="ContactLetterNav-letter"
    onClick={[Function]}
  >
    L
  </li>
  <li
    className="ContactLetterNav-letter"
    onClick={[Function]}
  >
    M
  </li>
  <li
    className="ContactLetterNav-letter"
    onClick={[Function]}
  >
    N
  </li>
  <li
    className="ContactLetterNav-letter"
    onClick={[Function]}
  >
    O
  </li>
  <li
    className="ContactLetterNav-letter"
    onClick={[Function]}
  >
    P
  </li>
  <li
    className="ContactLetterNav-letter"
    onClick={[Function]}
  >
    R
  </li>
  <li
    className="ContactLetterNav-letter"
    onClick={[Function]}
  >
    S
  </li>
  <li
    className="ContactLetterNav-letter"
    onClick={[Function]}
  >
    T
  </li>
  <li
    className="ContactLetterNav-letter"
    onClick={[Function]}
  >
    V
  </li>
  <li
    className="ContactLetterNav-letter"
    onClick={[Function]}
  >
    W
  </li>
  <li
    className="ContactLetterNav-letter"
    onClick={[Function]}
  >
    Y
  </li>
</ul>
`;
